### PR TITLE
Upgrade dart examples to use yaml v2 config file

### PR DESCRIPTION
**Pull request by andreia-oca**

### CODE REVIEW

File: dart/chatbot-openai/server/genezio.yaml [hunk 1]
```diff
@@ -1,16 +1,52 @@
...
+# It is optional. It can also be an array if there are multiple frontends you want to deploy.
 frontend:
-  path: ../client/build/web
-classes:
-  - path: lib/chat_backend.dart
-    type: jsonrpc
-    name: ChatBackend
-    methods: []
+  # The folder where the frontend scripts will run.
+  path: ../client/
```
Comment (costinsin) [inline dart/chatbot-openai/server/genezio.yaml:33]: You should move `genezio.yaml` in the root. We want to encourage the "workspace" setup.
```suggestion
  path: client
```

File: dart/chatbot-openai/server/genezio.yaml [hunk 2]
```diff
@@ -1,16 +1,52 @@
...
+  # Specifies information about the SDK generation. If not specified, the SDK will not be generated for this frontend.
+  sdk:
+    # The language the SDK will be written in. Usually the same as the frontend language.
+    language: dart
+    # The path where the SDK will be generated. Not required for languages that export the SDK to a package manager.
+    path: ./lib/sdk
```
Comment (costinsin) [inline dart/chatbot-openai/server/genezio.yaml:39]: I recommend not including `./` in paths.

```suggestion
    path: lib/sdk
```

File: dart/chatbot-openai/server/genezio.yaml [hunk 3]
```diff
@@ -1,16 +1,52 @@
...
+  # Classes that will be deployed
+  classes:
+    - name: ChatBackend
+      path: ./lib/chat_backend.dart
+      type: jsonrpc
+  # Scripts are running in the specified `path` directory.
+  scripts:
+    # List of scripts to run before deploying the backend.
+    deploy:
+      - dart pub get
```
Comment (costinsin) [inline dart/chatbot-openai/server/genezio.yaml:23]:
```suggestion
    deploy: dart pub get
```

File: dart/chatbot-openai/server/genezio.yaml [hunk 4]
```diff
@@ -1,16 +1,52 @@
...
+  path: .
+  # Information about the backend's programming language.
+  language:
+    # The name of the programming language.
+    name: dart
+  # Classes that will be deployed
+  classes:
+    - name: ChatBackend
+      path: ./lib/chat_backend.dart
+      type: jsonrpc
```
Comment (costinsin) [inline dart/chatbot-openai/server/genezio.yaml:18]: Maybe remove this?
```suggestion

```

**Review by costinsin — APPROVED** [review]
The comments I have left apply to all examples.

File: dart/chatbot-openai/server/genezio.yaml (hunk 1 above)
Comment (andreia-oca) [reply dart/chatbot-openai/server/genezio.yaml:33]: I tried to move `genezio.yaml` to workspaces and that yield a bunch of errors. 
I didn't continue to investigate the problem. For now, we are gonna put `genezio.yaml` in the backend directory.

**Pull request merged**